### PR TITLE
add(docs): add a guide about how to run on Mac

### PR DESCRIPTION
**Pull request by Fatpandac**

### CODE REVIEW

**Comment by Hefeweizen** [comment]
In executing on my mac, I found two issues:
1. `#!` is wrong.  I changed my copy to `/usr/bin/env bash`
2. mac date doesn't accept `-d` as input date.  Still working on that

**Comment by Hefeweizen** [comment]
grabbed gnu coreutils (`brew install coreutils`) and replaced instances of the command `date` with `gdate`. 

Next step, massive color bleeding...

**Comment by Fatpandac** [comment]
Follow this guide can't run the script on your Mac?

**Comment by Hefeweizen** [comment]
Yes, you covered everything.  I got lost in the interface; I thought I was reading an issue, rather than seeing this as a PR with suggestions.

**Comment by Fatpandac** [comment]
Add it into README, you can quickly set up this script on mac without looking for issues to get how to run on Mac.

**Comment by aaossa** [comment]
Hi, thank you both for your comments. As you may guess, I did not test this is in a Mac environment. I'm open to adding `coreutils` as requirement if needed, but I'm not sure if this would be the preferred approach for Mac users, any opinions on this? I was thinking of testing if the `gdate` command is available and use `date` as fallback:

```bash
# Use gdate if available (possibly on BSD-related environments)
if hash gdate 2>/dev/null; then
  function _date { gdate "$@" ;}
else
  function _date { date "$@" ;}
fi
```

This requires replacing every use of the `date` command in the script to use the `_date` function instead (6 times in 5 lines, not that much). Would this work? Also, can you provide a sample of the exception you're seeing if you don't have `coreutils` installed? Thank you both once again.

**Comment by Fatpandac** [comment]
It runs well with `coreutils`, but if don't have `coreutils` will get `illegal option -- d`.
Maybe you can add a output to tell Mac user install `coreutils`, like below:
```shell
if ! command -v gdate &> /dev/null; then
  echo "On mac you need install coreutils to run this script. (link to issue or docs)";
  exit 1;
fi
```

**Comment by aaossa** [comment]
> It runs well with `coreutils`, but if don't have `coreutils` will get `illegal option -- d`. Maybe you can add a output to tell Mac user install `coreutils`, like below:
> 
> ```shell
> if ! command -v gdate &> /dev/null; then
>   echo "On mac you need install coreutils to run this script. (link to issue or docs)";
>   exit 1;
> fi
> ```

Seems like a good idea, the only problem I see is that any non-mac environment will show that warning/exception. Any ideas on how to easily identify a mac environment? Maybe it will be easier to check if the `date` command maps to the GNU version. [This](https://stackoverflow.com/a/8748344/3281097) seems like a good approach:

```bash
if date --version >/dev/null 2>&1 ; then
    function _date { date "$@" ;}
else
    if ! command -v gdate &> /dev/null; then
      echo "On mac you need install coreutils to run this script. (link to issue or docs)";
      exit 1;
    else
      function _date { gdate "$@" ;}
    fi
fi
```

**Comment by Fatpandac** [comment]
No idea, this looks like so good.
Bash version also needs to check too, `declare -A` support begins at V4 (see [this](https://stackoverflow.com/a/43948526)), Mac default is V3.

**Comment by Fatpandac** [comment]
So, should I push a commit to add this code or you?

**Comment by aaossa** [comment]
> So, should I push a commit to add this code or you?

I could create a PR to add the behaviour I previously described and changing the shebang. Could you update your PR to explain that `coreutils` is a requirement on Mac? About `declare -A` support, I think another PR will be necessary

**Comment by Fatpandac** [comment]
I have already explained that `coreutils` is a requirement in [my PR](https://github.com/aaossa/git-activity/pull/1/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R64).

**Comment by aaossa** [comment]
> I have already explained that `coreutils` is a requirement in [my PR](https://github.com/aaossa/git-activity/pull/1/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R64).

I'm working on a new version of the command #5 that already includes the proposed shebang and the checks for `date` and `gdate`. Can you try out the PR and let me know if it works?

**Comment by Fatpandac** [comment]
It's fine on #5.

**Comment by aaossa** [comment]
Ok, #5 is merged now. Can you update your PR to only include the first step of the instructions you've made? The only real requirement now is `coreutils` on Mac. Any other change should not be necessary and is expected to be reported as a bug

**Comment by Fatpandac** [comment]
@aaossa done.

File: README.md [hunk 1]
```diff
@@ -57,6 +57,9 @@ Fri █████████████████████████
 
 ![image](https://user-images.githubusercontent.com/10425834/170083883-6145a11f-b6d8-490e-a751-45c2d6552698.png)
 
+### Run On Mac
+
+Just need install `coreutils`, execute cammond `brew install coreutils` if you had brew.
```
Comment (aaossa) [inline README.md]: Can we change this to something like:
```raw
You need to install `coreutils`. If you have brew available, you can execute the command `brew install coreutils`.
```

**Review by aaossa — CHANGES_REQUESTED** [review]
I requested a change to fix a couple of grammar mistakes

File: README.md (hunk 1 above)
Comment (Fatpandac) [reply README.md]: OK, thx.

**Review by aaossa — APPROVED** [review]
Thank you 🙌

**Comment by aaossa** [comment]
Thanks @Fatpandac for your PR and the comments that led to other changes in the script. Same goes to @Hefeweizen . Now Mac users should be able to use the script without further problems 👌 Awesome!

**Pull request merged**